### PR TITLE
Minor UI improvements and label changes

### PR DESCRIPTION
**Pull request by fenix2222**

### CODE REVIEW

File: StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.html [hunk 1]
```diff
@@ -17,30 +17,30 @@ <h5 class="modal-title">{{title}}</h5>
                 </div>
                 <div class="form-group clearfix" data-toggle="tooltip" data-placement="right" title="The amount of {{ coinUnit }} to send with this transaction">
                     <label for="amount">Amount</label>
-                    <input formControlName="amount" id="amount" type="number" step="0.01" class="form-control" placeholder="Amount" [class.is-invalid]="amount.invalid && (amount.dirty || amount.touched)">
+                    <input formControlName="amount" id="amount" type="number" step="0.01" class="form-control" placeholder="Amount" [class.is-invalid]="amount.invalid && (amount.dirty || amount.touched)" min="0">
```
Comment (dev0tion) [inline StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.html]: min=0 should actually be checked in transaction.component.ts using a validator like we do throughout the rest of the wallet.
This is beyond the scope of the current issue though, but needs changing throughout the entire smart contracts component.

File: StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.html [hunk 2]
```diff
@@ -17,30 +17,30 @@ <h5 class="modal-title">{{title}}</h5>
...
                     <div *ngIf="amount.errors" class="invalid-feedback">
-                        <p *ngIf="amount.errors.amountError">Amount must be less than your balance</p>
+                        <p *ngIf="amount.errors.amountError">The amount you have entered exceeds balance available at the sender address</p>
                     </div>
                 </div>
                 <div class="form-group clearfix" data-toggle="tooltip" data-placement="right" title="The fee to send with this transaction ({{ coinUnit }}/bytes)">
                     <label for="feeAmount">Fee</label>
-                    <input formControlName="feeAmount" id="feeAmount" type="number" step="0.01" class="form-control" placeholder="Fee" [class.is-invalid]="feeAmount.invalid && (feeAmount.dirty || feeAmount.touched)">
+                    <input formControlName="feeAmount" id="feeAmount" type="number" step="0.01" class="form-control" placeholder="Fee" [class.is-invalid]="feeAmount.invalid && (feeAmount.dirty || feeAmount.touched)" min="0">
```
Comment (dev0tion) [inline StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.html]: min=0 should actually be checked in transaction.component.ts using a validator like we do throughout the rest of the wallet.
This is beyond the scope of the current issue though, but needs changing throughout the entire smart contracts component.

File: StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.html [hunk 3]
```diff
@@ -17,30 +17,30 @@ <h5 class="modal-title">{{title}}</h5>
...
                     <div *ngIf="feeAmount.errors" class="invalid-feedback">
                         <p *ngIf="feeAmount.errors.feeError">Fee must be less than your balance</p>
                     </div>
                 </div>
                 <div class="row">
                     <div class="form-group col-6 clearfix" data-toggle="tooltip" data-placement="right" title="The gas price to use with this transaction ({{ coinUnit }}/gas)">
                         <label for="gasPrice">Gas price</label>
-                        <input formControlName="gasPrice" id="gasPrice" type="number" class="form-control" placeholder="Gas Price" [class.is-invalid]="gasPrice.invalid && (gasPrice.dirty || gasPrice.touched)">
+                        <input formControlName="gasPrice" id="gasPrice" type="number" class="form-control" placeholder="Gas Price" [class.is-invalid]="gasPrice.invalid && (gasPrice.dirty || gasPrice.touched)" min="0">
```
Comment (dev0tion) [inline StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.html]: min=0 should actually be checked in transaction.component.ts using a validator like we do throughout the rest of the wallet.
This is beyond the scope of the current issue though, but needs changing throughout the entire smart contracts component.

File: StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.html [hunk 4]
```diff
@@ -17,30 +17,30 @@ <h5 class="modal-title">{{title}}</h5>
...
                         <div *ngIf="gasPrice.errors" class="invalid-feedback">
                             <p *ngIf="gasPrice.errors.gasPriceTooLowError">Gas price must be greater than {{ gasPriceMinimum }}</p>
                             <p *ngIf="gasPrice.errors.gasPriceTooHighError">Gas price must be less than {{ gasPriceMaximum }}</p>
                         </div>
                     </div>
                     <div class="form-group col-6 clearfix" data-toggle="tooltip" data-placement="right" title="The gas limit to use with this transaction">
                         <label for="gasLimit">Gas limit</label>
-                        <input formControlName="gasLimit" id="gasLimit" type="number" class="form-control" placeholder="Gas Limit" [class.is-invalid]="gasLimit.invalid && (gasLimit.dirty || gasLimit.touched)">
+                        <input formControlName="gasLimit" id="gasLimit" type="number" class="form-control" placeholder="Gas Limit" [class.is-invalid]="gasLimit.invalid && (gasLimit.dirty || gasLimit.touched)" min="0">
```
Comment (dev0tion) [inline StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.html]: min=0 should actually be checked in transaction.component.ts using a validator like we do throughout the rest of the wallet.
This is beyond the scope of the current issue though, but needs changing throughout the entire smart contracts component.

File: StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.ts [hunk 5]
```diff
@@ -141,9 +140,9 @@ export class TransactionComponent implements OnInit {
 
         let gasLimitValidator = (this.mode === Mode.Call ? gasCallLimitMinimumValidator : gasCreateLimitMinimumValidator);
 
-        this.amount = new FormControl(0, [amountValidator]);
+        this.amount = new FormControl(0.1, [amountValidator]);
```
Comment (dev0tion) [inline StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.ts]: Why is this set to 0.1 per default? Not too familiar with the smart contracts so I'm just wondering.

File: StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.html (hunk 1 above)
Comment (dev0tion) [reply StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.html]: I will add a separate issue for this.

File: StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.ts (hunk 5 above)
Comment (fenix2222) [reply StratisCore.UI/src/app/wallet/smart-contracts/components/modals/transaction/transaction.component.ts]: Maciej comment was: `Contract deployment or call with amount 0 will not execute`, so basically we are just putting a meaningful default

**Pull request merged**